pipit/tests/trace.py

```diff
@@ -97,5 +103,8 @@ def test_match_caller_callee(data_dir, ping_pong_otf2_trace):
 
     df = trace.events
 
-    # all events of the ping pong trace are roots with no children
-    assert len(df.loc[(df["Event Type"] == "Enter") & (df["_parent"].notnull())]) == 0
+    # nodes with a parent = 40
+    assert len(df.loc[df["_parent"].notnull()]) == 40
+
+    # nodes with children = 2
+    assert len(df.loc[df["_children"].notnull()]) == 2
```

Comment:
I meant something like:

`assert len(df.loc[len(df["_children"]) == 0]) == 40`
`assert len(df.loc[len(df["_children"]) != 0]) == 2`

Reply:
ok

Reply:
@bhatele I think that is the simplest expression possible

to do something as you mentioned, we would have to do
enter_df = df.loc[df["Event Type"] == "Enter"]
assert list(enter_df.apply(lambda row: 0 if row["_children"] is None else len(row["_children"]), axis = 1)).count(0) == 40

because we have to filter to enter rows only (to not count instant events) and then also handle the case for leaf functions where the children is None rather than an empty list